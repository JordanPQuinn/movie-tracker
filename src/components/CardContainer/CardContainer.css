.CardContainer {
  padding-top: 30px;
}