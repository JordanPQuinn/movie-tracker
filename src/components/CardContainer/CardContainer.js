import React from 'react';

export const CardContainer = (props) => {
  return (
    <div>
      <h1> container </h1>
    </div>
  )
}