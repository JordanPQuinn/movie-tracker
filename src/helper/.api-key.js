export const key = '030166c9615690cd48d154e971eb528a'

export const url = 'https://api.themoviedb.org/3/discover/movie?api_key=030166c9615690cd48d154e971eb528a&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1'