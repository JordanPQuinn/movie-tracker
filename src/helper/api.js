export const fetchApi = async (url) => {
  const initalFetch = await fetch(url);
  const resolvedPromise = await initalFetch.json();
  return resolvedPromise
};

export const getFilms = async (url) => {
  const { results } = await fetchApi(url);
  return cleanFilms(results)
}

const cleanFilms = films => {
  return films.map(film => ({
    backdrop: film.backdrop_path,
    title: film.title,
    overview: film.overview,
    poster: film.poster_path,
    date: film.release_date,
    rating: film.vote_average,
    id: film.id
  }));
};

export const getUserData = async (url, state) => {
  const { data } = await fetchApi(url)
  const { email, password } = state;
  const user = data.find(user => {
    return user.email === email && user.password === password;
  });
  const currentFavorites = await fetchApi(`${url}/${user.id}/favorites/`)
  user.favorites = [...currentFavorites.data]
  return user;
}
