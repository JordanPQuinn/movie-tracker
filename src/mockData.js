export const mockData = {
  mockUser: {
    name: 'TinkerBell',
    email: 'tink@gmail.com',
    password: 'password'
  },
  mockFilmsArray: [
    {
      backdrop: 'url',
      title: 'Planet Earth',
      overview: 'Best documentary ever',
      poster_path: 'url',
      release_date: 'March 5, 2006',
      vote_average: '1000%',
      movie_id: 20
    }
  ],
  mockEvent: {
    target: { 
      id: 0,
      name: 'email',
      value: 'sickemail' 
    },
    preventDefault: () => {}
  },
  mockFavoritesArray: [  
    {
      backdrop: 'url',
      title: 'Planet Earth 2',
      overview: 'Also best documentary ever',
      poster_path: 'url',
      release_date: 'September 15, 2017',
      vote_average: '1000%',
      movie_id: 21
    },    
    {
      backdrop: 'url',
      title: 'Blue Planet',
      overview: 'Awesome but not best',
      poster_path: 'url',
      release_date: 'April 4, 2007',
      vote_average: '400%',
      movie_id: 24
    }
  ],
  mockFavoritesIdOnly: [
    { movie_id: 1 }, 
    { movie_id: 2 }
  ],
  mockFilmsIdOnly: [
    { movie_id: 0 }, 
    { movie_id: 1 }, 
    { movie_id: 2 }
  ],
  loginDefaultState: {
    email: '',
    password: ''
  },
  mockSignupDefaultState: {
    name: '',
    error: false,
    password: '',
    email: ''
  },
  mockLoginExpectedState: {
    email: 'sickemail',
    password: ''
  },
  mockSignupExpectedState: {
    email: "sickemail", 
    error: false, 
    name: "", 
    password: ""
  },
  mockRawFilmData: {
    "vote_count": 6793,
    "id": 198663,
    "video": false,
    "vote_average": 7,
    "title": "The Maze Runner",
    "popularity": 535.445142,
    "poster_path": "/coss7RgL0NH6g4fC2s5atvf3dFO.jpg",
    "original_language": "en",
    "original_title": "The Maze Runner",
    "genre_ids": [
    28,
    9648,
    878,
    53
    ],
    "backdrop_path": "/lkOZcsXcOLZYeJ2YxJd3vSldvU4.jpg",
    "adult": false,
    "overview": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
    "release_date": "2014-09-10"
  },
  mockCleanedFilmData: {
    backdrop: "/lkOZcsXcOLZYeJ2YxJd3vSldvU4.jpg",
    title: "The Maze Runner",
    overview: "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
    poster_path: "/coss7RgL0NH6g4fC2s5atvf3dFO.jpg",
    release_date: "2014-09-10",
    vote_average: 7,
    movie_id: 198663
  },
  mockFavoriteToStore: {
    movie_id: 2,
    user_id: 10,
    title: '',
    poster_path: '',
    release_date: '',
    vote_average: '',
    overview: ''
  }
}

