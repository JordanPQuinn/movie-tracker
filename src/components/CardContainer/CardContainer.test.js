/*eslint-disable*/
import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps, mapDispatchToProps } from './CardContainer';
import * as api from '../../helper/api';
import { mockData } from '../../mockData';

describe('CardContainer', () => {
  it('should match the snapshot', () => {
    const renderedCardContainer = shallow(<CardContainer films={mockData.mockFilmsArray} />);

    expect(renderedCardContainer).toMatchSnapshot();
  });

  it('should correctly map the store', () => {
    const mapped = mapStateToProps({ user: mockData.mockUser });

    expect(mapped.user).toEqual(mockData.mockUser);
  });

  it('should call the dispatch fn when using a fn from MDTP', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.addFavorite(mockData.mockFilmsArray[0]);
    expect(mockDispatch).toHaveBeenCalled();

    mapped.removeFavorite(mockData.mockFilmsArray[0]);
    expect(mockDispatch).toHaveBeenCalled();
  });

  describe('handleFavorite', () => {
    it('should call handleAddFav with the correct args if film is not in user favorites', () => {
      const mockUser = { favorites: mockData.mockFavoritesIdOnly };
      const mockAddFav = jest.fn();

      const renderedCardContainer = shallow(
        <CardContainer
          user={ mockUser }
          films={ mockData.mockFilmsIdOnly }
          addFavorite={ mockAddFav }
        />
      );

      renderedCardContainer.instance().handleAddFavorite = jest.fn();

      renderedCardContainer.instance().handleFavorite(0);
      expect(
        renderedCardContainer.instance().handleAddFavorite
      ).toHaveBeenCalledWith(0);
    });

    it('should call handleRemoveFav with the correct args if film is in user favorites', () => {
      const mockUser = { favorites: mockData.mockFilmsIdOnly };
      const mockRemoveFav = jest.fn();

      const renderedCardContainer = shallow(
        <CardContainer
          user={ mockUser }
          films={ mockData.mockFilmsIdOnly }
          removeFavorite={ mockRemoveFav }
        />
      );

      renderedCardContainer.instance().handleRemoveFavorite = jest.fn();

      renderedCardContainer.instance().handleFavorite(0);
      expect(
        renderedCardContainer.instance().handleRemoveFavorite
      ).toHaveBeenCalledWith(mockData.mockFilmsIdOnly[0]);
    });
  });

  describe('handleAddFavorite', () => {
    it('should call addFavorite and sendFavorite with correct args', () => {
      const mockUser = { favorites: mockData.mockFavoritesIdOnly };
      const mockAddFav = jest.fn();
      api.sendFavorite = jest.fn();

      const renderedCardContainer = shallow(
        <CardContainer
          user={mockUser}
          films={mockData.mockFilmsIdOnly}
          addFavorite={mockAddFav}
        />
      );

      renderedCardContainer.instance().handleAddFavorite(0);
      expect(mockAddFav).toHaveBeenCalledWith(mockData.mockFilmsIdOnly[0]);
      expect(api.sendFavorite).toHaveBeenCalledWith(mockUser, mockData.mockFilmsIdOnly[0]);
    });
  });

  describe('handleRemoveFavorite', () => {
    it('should call removeFavorite and deleteFavorite with the correct args', () => {
      const mockUser = { favorites: mockData.mockFavoritesIdOnly };
      const mockRemoveFav = jest.fn();
      api.deleteFavorite = jest.fn();

      const renderedCardContainer = shallow(
        <CardContainer
          user={mockUser}
          films={mockData.mockFilmsIdOnly}
          removeFavorite={mockRemoveFav}
        />
      );

      renderedCardContainer.instance().handleRemoveFavorite(mockData.mockFilmsIdOnly[1]);
      expect(mockRemoveFav).toHaveBeenCalledWith(mockData.mockFilmsIdOnly[1]);
      expect(api.deleteFavorite).toHaveBeenCalledWith(mockUser, mockData.mockFilmsIdOnly[1]);
    });
  });
});
