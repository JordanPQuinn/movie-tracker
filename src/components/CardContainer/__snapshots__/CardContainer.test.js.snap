// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    films={
        Array [
            Object {
              "backdrop": "url",
              "movie_id": 20,
              "overview": "Best documentary ever",
              "poster_path": "url",
              "release_date": "March 5, 2006",
              "title": "Planet Earth",
              "vote_average": "1000%",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          film={
                    Object {
                              "backdrop": "url",
                              "movie_id": 20,
                              "overview": "Best documentary ever",
                              "poster_path": "url",
                              "release_date": "March 5, 2006",
                              "title": "Planet Earth",
                              "vote_average": "1000%",
                            }
          }
          handleFavorite={[Function]}
          user={undefined}
/>,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "film": Object {
            "backdrop": "url",
            "movie_id": 20,
            "overview": "Best documentary ever",
            "poster_path": "url",
            "release_date": "March 5, 2006",
            "title": "Planet Earth",
            "vote_average": "1000%",
          },
          "handleFavorite": [Function],
          "user": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            film={
                        Object {
                                    "backdrop": "url",
                                    "movie_id": 20,
                                    "overview": "Best documentary ever",
                                    "poster_path": "url",
                                    "release_date": "March 5, 2006",
                                    "title": "Planet Earth",
                                    "vote_average": "1000%",
                                  }
            }
            handleFavorite={[Function]}
            user={undefined}
/>,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "film": Object {
              "backdrop": "url",
              "movie_id": 20,
              "overview": "Best documentary ever",
              "poster_path": "url",
              "release_date": "March 5, 2006",
              "title": "Planet Earth",
              "vote_average": "1000%",
            },
            "handleFavorite": [Function],
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
